### PR TITLE
Re-enable Sauce

### PR DESCRIPTION
**Pull request by dlockhart**
This re-enables SauceLabs testing for core. I've changed the Travis CI setup so that BrowserStack & Sauce tests happen in parallel in separate jobs, which will make it easier to directly compare their runtimes.

Sauce tests will always "pass" (be green), so if they fail they'll never prevent us from merging. We'll need to be careful though when doing any comparison analysis since Sauce could fail early and appear faster than BrowserStack.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,22 +25,35 @@ addons:
...
       env:
         - BROWSERSTACK_USERNAME: margareepeacocke1
         # BROWSERSTACK_ACCESS_KEY
         - secure: LLdLul0G53je1zR+Mkn3KGJg3PXco6cgHcjElAJS1ZlA8bT2TGLmvdv3rAbqKG6Ff1LOtEViFJxXT/A31mu2wxEZl0xcZE4Vu17Ykz9n9SOl0UIFyBWruKj0g7H34eQyKL1yBZ5XUuiE5qLgJSepyrYF4v25X3cwojx4HEDw88bGzTJLuazPS9SZc5FspP/EFgvkXTGC/FWqEURDIJtZedvtLlGWsb/dZ6O2d8EGhZFYtnDNPSUhUSC2t/xkwxd/+zkj7ooci1cWBNrETBGKFhSKHdA46FyPz437siDa4nG5KL8an9wVND9sZ1LaF/CC1zDA902iY9SxP5rrRehrBqaQg+M3GOxs2yTJJqNM7fm7ws5MHKeaF2pukXkj+CR6vCOIV/2sGCOa0lwFGrV7EQ77oeVKS1q6iSz/5QJNGIppAoNKRSE604Pl7l7WEyUY4KJ/TVtWiPSTtpn4SMmpofIF3N0qtUIy2cXb+6MVhp7Tgo4MeErWKjplMdXC9ZkGtGw407wKX6oCtof74kCzkWQ3W+gAc38/Uvz7P1ky/vbxz4OPmBNL8cCBFnWRciWO8P5rcKgWHLGjkn7CMwYwjSu8QwQW+aWA/9c0sA98SoVz9TUEnG3l/ZJozkSpzH7n+T8iwY+T8F3tXPn6Ycdjrip/N6A/LKy7rSZi+zbQvKY=
+    - script:
+      - |
+        if [ $TRAVIS_PULL_REQUEST != false ] && [ $TRAVIS_SECURE_ENV_VARS == true ]; then
+            echo "Pull request with secure environment variables, running Sauce tests...";
+            npm run build;
+            npm run test:sauce || true;
```
[the remark this answers is not in the document]
Comment (margaree) [reply .travis.yml:53]: What will this do when it's not in a PR? Just skip it?

File: karma.sauce.conf.js [hunk 2]
```diff
@@ -34,8 +34,10 @@ const customLaunchers = {
 };
 
 module.exports = config => {
+	const defaultConfig = createDefaultConfig(config);
+	defaultConfig.browsers = []; // remove ChromeHeadless
```
[the remark this answers is not in the document]
Comment (margaree) [reply karma.sauce.conf.js:38]: Looks like with browserstack the headless tests add about 5 - 6 seconds, so this wouldn't really affect comparisons

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dlockhart) [reply .travis.yml:53]: Yeah exactly, same as before and same as the BrowserStack tests. If it's not a pull request or if it's in a fork it'll do nothing. The BrowserStack ones will run headless in that case.

**Review by margaree — APPROVED** [review]

File: karma.sauce.conf.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlockhart) [reply karma.sauce.conf.js:38]: Good call though -- I've removed it from BrowserStack too to make the comparisons more accurate.

**Pull request merged**